### PR TITLE
Fix assessing new models when retrieving them for assessment

### PR DESCRIPTION
**Pull request by iozbeyli**
### Motivation and Context
When a student submits a modeling submission after the remaining submissions assessed, Compass was not suggesting automatic assessments on submissions page of exercise. This feature was working on assessment dashboard but not submissions page.

### Steps for Testing
1. Add a submission
2. Manually assess it
3. Create a new submission
4. Go to exercise/submissions
5. Assess model
6. Suggestions must appear

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java [hunk 1]
```diff
@@ -529,10 +529,12 @@ public void printStatistic(Long exerciseId) {
      */
     public List<Long> getCalculationEngineModelsWaitingForAssessment(Long exerciseId) {
         List<ModelingSubmission> modelingSubmissions = modelingSubmissionRepository.findSubmittedByExerciseIdWithEagerResultsAndFeedback(exerciseId);
-        assessAllAutomatically(modelingSubmissions.stream().map(Submission::getId).collect(Collectors.toList()), exerciseId);
+
         CompassCalculationEngine engine = compassCalculationEngines.get(exerciseId);
         engine.notifyNewModels(modelingSubmissions);
 
+        assessAllAutomatically(modelingSubmissions.stream().map(Submission::getId).collect(Collectors.toList()), exerciseId);
```
Comment (jamalxo) [inline src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java:536]: why is the ordering of importance here?

**Review by jamalxo — COMMENTED** [review]
tests are also still failing

File: src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java (hunk 1 above)
Comment (iozbeyli) [reply src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java:536]: Engine needs to be notified about the new models before assessing them, there is a preparation step before assessment.

**Comment by iozbeyli** [comment]
I couldn't make it fail locally, I will look further into it. However, another test is failing locally, getResultsForCurrentStudent_assessorHiddenForStudent for text submissions, which is unrelated.

**Review by SimonEntholzer — APPROVED** [review]
Works as intended :+1:

**Review by coolchock — APPROVED** [review]

**Review by PatZintel — APPROVED** [review]

**Review by jamalxo — APPROVED** [review]

**Comment by iozbeyli** [comment]
I need to check for flaky tests but worked as expected in test session as well

**Review by imstreetspirit — APPROVED** [review]

**Comment by iozbeyli** [comment]
I believe flaky tests are not related to this.

**Review by krusche — APPROVED** [review]

**Pull request merged**